src/components/patch/PatchContainer.tsx

```diff
@@ -80,6 +88,17 @@ export class PatchContainer extends React.Component<Props, State> {
     };
   }
 
+  public componentDidUpdate(prevProps: Props) {
```

Comment:
i think i understand why you used `componentDidUpdate` -- so that any changes to the url are captured and responded to? however i believe how the app currently works is that a url must be entered that either has the `USER_PAGE` or `PROJECT_PAGE` params. According to the [docs](https://reactjs.org/docs/react-component.html#componentdidupdate), `componentDidUpdate` is not called after initial render, which is when this logic would occur. therefore we want to perform this logic on `componentDidMount`. 

Am i missing some details or am I right by thinking that nothing in the app can toggle the params of the url that would require performing login on `didUpdate`. Entering a new URL will always cause a rerender of the entire app; therefore doing this logic in `componentDidMount` is sufficient.

Reply:
good point, we could change the componentDidUpdate to componentDidMount. You're right that we cannot toggle the params from the app and right now the componentDidUpdate is unnecessary.